etcdctl/ctlv3/command/role_command.go

```diff
@@ -41,6 +49,21 @@ func newRoleAddCommand() *cobra.Command {
 	}
 }
 
+func newRoleGrantCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "grant <role name> <key or key prefix>",
+		Short: "grant a role to a user",
+		Run:   roleGrantCommandFunc,
+	}
+
+	cmd.Flags().StringVar(&key, "key", "", "a key granted to to a role")
+	cmd.Flags().BoolVar(&prefix, "prefix", false, "a key is used as a prefix")
```

Comment:
i would image this command to be <role name> <permission type> <key>

```
etcdctl role grant role_foo read key_foo
```

or 

```
etcdctl role grant role_foo read prefix_foo --prefix
```
for prefix.